src/components/quiz-card/QuizCard.tsx

```diff
@@ -1,23 +1,42 @@
 import "../category-card/categoryCard.css";
-import { Link } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
 import { QuizCardProp } from "../../types";
+import { useGame } from "../../hooks";
 
 const QuizCard = ({ quiz }: QuizCardProp) => {
-  const { quizName, quizDescription, quizStatus, quizImage } = quiz;
+  const { id, quizName, quizDescription, quizStatus, quizImage } = quiz;
+  const { getQuestions } = useGame();
+  const navigate = useNavigate();
+
+  const handlePlayNow = async () => {
+    try {
+      await getQuestions(id, quizName);
+      navigate("/rules");
+    } catch (err) {
+      console.log(err);
```

Comment:
You can use `console.error` here

Reply:
sure i will add proper error handling here in next pr